From dd5e082250ca7b45ed22f713e10f0ba78b10b798 Mon Sep 17 00:00:00 2001
From: Oliver Reiche <oliver.reiche@huawei.com>
Date: Thu, 29 Jun 2023 16:53:50 +0200
Subject: [PATCH 4/7] x86 musl support

Backport from GCC 6.1.0
  Git commit: cf57e99350d0b44876979c71986d8ca29c361f83
  Subversion revision: r223218

Original description:
  * config/i386/linux.h (MUSL_DYNAMIC_LINKER): Define.
  * config/i386/linux64.h (MUSL_DYNAMIC_LINKER32): Define.
  (MUSL_DYNAMIC_LINKER64, MUSL_DYNAMIC_LINKERX32): Define.

Co-authored-by: Szabolcs Nagy <szabolcs.nagy@arm.com>
Co-authored-by: Gregor Richards <gregor.richards@uwaterloo.ca>
---
 gcc/config/i386/linux.h   | 3 +++
 gcc/config/i386/linux64.h | 7 +++++++
 2 files changed, 10 insertions(+)

diff --git a/gcc/config/i386/linux.h b/gcc/config/i386/linux.h
index 73681fe82..f907adfb1 100644
--- a/gcc/config/i386/linux.h
+++ b/gcc/config/i386/linux.h
@@ -22,3 +22,6 @@ along with GCC; see the file COPYING3.  If not see
 
 #define GNU_USER_LINK_EMULATION "elf_i386"
 #define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.2"
+
+#undef MUSL_DYNAMIC_LINKER
+#define MUSL_DYNAMIC_LINKER "/lib/ld-musl-i386.so.1"
diff --git a/gcc/config/i386/linux64.h b/gcc/config/i386/linux64.h
index 5b0a212ce..cf361b264 100644
--- a/gcc/config/i386/linux64.h
+++ b/gcc/config/i386/linux64.h
@@ -31,3 +31,10 @@ see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
 #define GLIBC_DYNAMIC_LINKER32 "/lib/ld-linux.so.2"
 #define GLIBC_DYNAMIC_LINKER64 "/lib64/ld-linux-x86-64.so.2"
 #define GLIBC_DYNAMIC_LINKERX32 "/libx32/ld-linux-x32.so.2"
+
+#undef MUSL_DYNAMIC_LINKER32
+#define MUSL_DYNAMIC_LINKER32 "/lib/ld-musl-i386.so.1"
+#undef MUSL_DYNAMIC_LINKER64
+#define MUSL_DYNAMIC_LINKER64 "/lib/ld-musl-x86_64.so.1"
+#undef MUSL_DYNAMIC_LINKERX32
+#define MUSL_DYNAMIC_LINKERX32 "/lib/ld-musl-x32.so.1"
-- 
2.30.2

